### ISSUE TITLE
B.2.6.100 - Basic Privileges

### ISSUE THREAD

**Issue by aldefouw** [4bbakers/redcap_rsvc#39]
Feature: [B.2.6.100 - Basic Privileges](https://github.com/4bbakers/redcap_rsvc/tree/v14.7.0/./Feature%20Tests/B/Assign%20User%20Rights_6/B.2.6.100%20-%20Basic%20Privileges.feature)

**Comment by schatto1** [comment]
Once RCTF PR [#12](https://github.com/aldefouw/rctf/pull/12) and [#13](https://github.com/aldefouw/rctf/pull/13) are merged, this test should pass

**Comment by aldefouw** [comment]
@schatto1  - Sorry for the delay in review.   I've been meeting trapped today.  I'll hopefully get these reviews done now.

**Comment by mmcev106** [comment]
@schatto1, cypress run 1956 includes both rctf PRs.  However, [this feature still seems to be failing](https://cloud.cypress.io/projects/pvu79o/runs/1956/test-results?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%7B%22label%22%3A%22redcap_rsvc%2FFeature%20Tests%2FB%2FAssign%20User%20Rights_6%2FB.2.6.100%20-%20Basic%20Privileges.feature%22%2C%22value%22%3A%22%5B%5C%224361c0da-77cf-478a-a5b2-cdb614cf8e38%5C%22%5D%22%7D%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D).  Do you know why that might be?

**Comment by aldefouw** [comment]
@mmcev106  - I added in the fixes that @schatto1 requested in https://github.com/aldefouw/rctf/pull/13 for 

[RCTF v1.0.103](https://github.com/aldefouw/rctf/commit/0c54658c8d77147ae9ae12a65ac795a024379cae).

It looks like you're saying this isn't passing yet.

**Comment by mmcev106** [comment]
Correct.  One or both of the PRs are still a big step in the right direction, as more specs are passing now.  This particular one looks like it needs additional changes though.

**Comment by aldefouw** [comment]
@mmcev106 Gotcha.  

@schatto1  - I'll try running locally to see what I can discern.

**Comment by schatto1** [comment]
@aldefouw @mmcev106, this particular feature is depend on [RCTF PR #12](https://github.com/aldefouw/rctf/pull/12), that would fix the "No Access" and subsequent User Rights radio buttons not being selected properly, but that PR needed further review since I removed a `.parent()`

**Comment by aldefouw** [comment]
@schatto1  - To get you unblocked on this until I can fix the proper step definition, I recommend substituting [this step](https://github.com/4bbakers/redcap_rsvc/blob/v14.7.0/Feature%20Tests/B/Assign%20User%20Rights_6/B.2.6.100%20-%20Basic%20Privileges.feature#L30)

for this:

`And I check the radio labeled "No Access" in the dialog box`

Radios can be selected by a labeled near them even if it does't make sense in context of this test since the radios exist underneath the "User Rights" subsection.

This is not great option from a descriptive standpoint, but I think it'll work to get you unblocked.  I'll have to figure out a more robust fix within RCTF for the original step definition so it makes sense to a human reader.

**Comment by mmcev106** [comment]
@schatto1, you may want to review past comments, specifically [this one](https://github.com/4bbakers/redcap_rsvc/issues/39#issuecomment-2546894958) showing this feature still failing AFTER including RCTF PRs 12 & 13.  What do you think of [aldefouw's suggestion](https://github.com/4bbakers/redcap_rsvc/issues/39#issuecomment-2547132571)?

**Comment by schatto1** [comment]
@mmcev106 - @aldefouw 's solution worked, so I created a [new PR](https://github.com/4bbakers/redcap_rsvc/pull/162) with the changes. Once that is verified/merged, this issue should be done. Thanks all!

**Comment by mmcev106** [comment]
@schatto1, this is passing for me.  I've merged the PR and closed this issue.  @aldefouw thanks for the help!
